### PR TITLE
HZN-1475: Extend topology generator and test suite to support bridge topology

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA: https://issues.opennms.org/browse/HZN-1475

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

@j-white , @rssntn67, could you take another look? Thanks!

What is in this PR?
1. Enhancement of the topology generator by the protocol "bridge". The bridge protocol is a bit different in that it contains multiple node types. The generator builds a tree consisting of hosts, bridges, segments and" Mac/IpAdresses on Segment without node". The tree is repeated depending on the amount of nodes (multiple of 10).
2. The generated tree is tested in LinkdTopologyProviderTestIT.testBridge()
3. I introduced a "BridgeBuilder" that allows for easier generating of such topologies. It is sufficient for my use case but could be enhanced in the future. It tries to abstract the low level stitching together of the different database entries.

Thanks again to @rssntn67 who provided me with a working example and lots of explanation.

### CODE REVIEW

File: features/enlinkd/generator/src/main/java/org/opennms/enlinkd/generator/TopologyGenerator.java [hunk 1]
```diff
@@ -65,7 +66,7 @@ public static TopologyGeneratorBuilder builder() {
     }
 
     public enum Protocol {
-        cdp, isis, lldp, ospf
+        cdp, isis, lldp, ospf, bridgeBridge
```
Comment (j-white) [inline features/enlinkd/generator/src/main/java/org/opennms/enlinkd/generator/TopologyGenerator.java]: Can we just call it `bridge`?

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/test/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProviderTestIT.java [hunk 2]
```diff
@@ -143,6 +152,12 @@ public void testOspf() throws Exception {
         test(TopologyGenerator.Protocol.ospf);
     }
 
+    @Test
+    @Transactional
+    public void testBridgeBridge() throws Exception {
+        test(TopologyGenerator.Protocol.bridgeBridge);
```
Comment (j-white) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/test/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProviderTestIT.java]: Can we add more specific tests? i.e. verifying that specific nodes are connected via a given shared segment

File: features/enlinkd/generator/src/main/java/org/opennms/enlinkd/generator/TopologyGenerator.java (hunk 1 above)
Comment (patrick-schweizer) [reply features/enlinkd/generator/src/main/java/org/opennms/enlinkd/generator/TopologyGenerator.java]: sure. done.

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/test/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProviderTestIT.java (hunk 2 above)
Comment (patrick-schweizer) [reply features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/test/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProviderTestIT.java]: yes will do once we are able to actually generate the topology - unfortunately I am not there yet...

Comment (patrick-schweizer) [reply features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/test/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProviderTestIT.java]: done.

**Comment by j-white** [comment]
Can we target this to the `release-24.0.0` branch?

**Comment by patrick-schweizer** [comment]
> Can we target this to the `release-24.0.0` branch?

done: https://github.com/OpenNMS/opennms/pull/2390

**Pull request closed without merging**